### PR TITLE
feat: Web Modeler API v1

### PR DESCRIPTION
**Pull request by wollefitz**
Closes https://github.com/camunda/web-modeler/issues/6161

### CODE REVIEW

File: web-modeler-openapi-cli/README.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ Downloads all files of the project.
   overwrite each other and the last file wins
 * `CANONICAL_PATH` - Recommended if your folder and file names are not unique. It uses the canonical path. Each folder
   and file name is suffixed by its unique ID (i.e.. `<name>[<id>]`). This makes names unique, but also less pleasant to
-  look at)
+  look at
```
Comment (1nb0und) [inline web-modeler-openapi-cli/README.md:58]:
```suggestion
  look at.
```

File: web-modeler-openapi-cli/README.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ Downloads all files of the project.
   overwrite each other and the last file wins
 * `CANONICAL_PATH` - Recommended if your folder and file names are not unique. It uses the canonical path. Each folder
   and file name is suffixed by its unique ID (i.e.. `<name>[<id>]`). This makes names unique, but also less pleasant to
```
Comment (1nb0und) [inline web-modeler-openapi-cli/README.md:57]:
```suggestion
  and file name is suffixed by its unique ID (i.e. `<name>[<id>]`). This makes names unique, but also less pleasant to
```

File: web-modeler-openapi-cli/docs/CollaboratorsApi.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+| **404**     | Not Found             | -                |
+| **500**     | Internal Server Error | -                |
+
+<a name="modifyCollaborator"></a>
+
+# **modifyCollaborator**
+
+> modifyCollaborator(createCollaboratorDto)
+
+Adds a new collaborator to a project or modifies the permission level of an existing collaborator.&lt;br/&gt;&lt;br/&gt; &lt;strong&gt;Note:&lt;/strong&gt; Only users that are part of the authorized organization (see &lt;code&gt;GET /api/v1/info&lt;/code&gt;) &lt;i&gt;and&lt;/i&gt; logged in to Web Modeler at least once can be added to a project.
```
Comment (1nb0und) [inline web-modeler-openapi-cli/docs/CollaboratorsApi.md:89]: For some reason the html syntax doesn't render properly. Is this intentional? I think we might want to find the appropriate markdown syntax

File: web-modeler-openapi-cli/docs/CollaboratorsApi.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+| **404**     | Not Found             | -                |
+| **500**     | Internal Server Error | -                |
+
+<a name="searchCollaborators"></a>
+
+# **searchCollaborators**
+
+> PubSearchResultDtoProjectCollaboratorDto searchCollaborators(pubSearchDtoProjectCollaboratorDto)
+
+Searches for collaborators.&lt;br/&gt; &lt;ul&gt;   &lt;li&gt;&lt;p&gt;&lt;em&gt;filter&lt;/em&gt; specifies which fields should match. Only items that match the given fields will be    returned.&lt;/p&gt;&lt;/li&gt;   &lt;li&gt;&lt;p&gt;&lt;em&gt;sort&lt;/em&gt; specifies by which fields and direction (&lt;code&gt;ASC&lt;/code&gt;/&lt;code&gt;DESC&lt;/code&gt;) the result    should be sorted.&lt;/p&gt;&lt;/li&gt;   &lt;li&gt;&lt;p&gt;&lt;em&gt;page&lt;/em&gt; specifies the page number to return.&lt;/p&gt;&lt;/li&gt;   &lt;li&gt;&lt;p&gt;&lt;em&gt;size&lt;/em&gt; specifies the number of items per page. The default value is 10.&lt;/p&gt;&lt;/li&gt; &lt;/ul&gt;
```
Comment (1nb0und) [inline web-modeler-openapi-cli/docs/CollaboratorsApi.md:160]: Same with this as above

File: web-modeler-openapi-cli/src/main/java/org/camunda/community/webmodeler/cli/ListProjectsCommand.java [hunk 5]
```diff
@@ -2,16 +2,27 @@
...
+            names = {"--page", "-pg"},
+            description = "page number of the results to fetch, defaults to 0",
+            scope = CommandLine.ScopeType.LOCAL)
+    private int page;
+
     @Override
     public Integer call() throws Exception {
-        List<ProjectMetadataDto> projects = parent.buildProjectsClient().listProjects();
+        PubSearchDtoProjectMetadataDto searchDto =
+                new PubSearchDtoProjectMetadataDto().size(50).page(page);
```
Comment (1nb0und) [inline web-modeler-openapi-cli/src/main/java/org/camunda/community/webmodeler/cli/ListProjectsCommand.java:23]: Might be better to extract `50` as a constant somewhere as it seems shared and we are defaulting to that amount for `ListProjectsCommand`, `ListFilesCommand` and `Downloader`

**Review by 1nb0und — COMMENTED** [review]
some minor feedback

File: web-modeler-openapi-cli/docs/CollaboratorsApi.md (hunk 3 above)
Comment (wollefitz) [reply web-modeler-openapi-cli/docs/CollaboratorsApi.md]: This is how it gets generated by `openapi-generator-maven-plugin` - for the other files I manually fixed. the broken HTML tags, I simply forgot it here.

We might want to look at how to fix this in a sustainable way in the future.

File: web-modeler-openapi-cli/docs/CollaboratorsApi.md (hunk 4 above)
Comment (wollefitz) [reply web-modeler-openapi-cli/docs/CollaboratorsApi.md]: See https://github.com/camunda-community-hub/web-modeler-java-client/pull/19#discussion_r1340501323

File: web-modeler-openapi-cli/src/main/java/org/camunda/community/webmodeler/cli/ListProjectsCommand.java (hunk 5 above)
Comment (wollefitz) [reply web-modeler-openapi-cli/src/main/java/org/camunda/community/webmodeler/cli/ListProjectsCommand.java]: Yep, makes sense!

**Review by 1nb0und — APPROVED** [review]

**Pull request merged**